src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -86,10 +94,47 @@ public string SearchText
             set
             {
                 searchText = value;
-                OnSearchTextChanged(this, EventArgs.Empty);
+                // The searchText is set multiple times before the control becomes visible and interactable.
+                // To prevent any debounces from triggering at some unexpected point before or after the control
+                // becomes visible, this flag is only set once the searchText value is set by the user
+                // (unless it somehow gets set somewhere else)
+                searchInitialized = searchInitialized || !string.IsNullOrEmpty(searchText);
```

Comment:
why not just run the debouncer if (!string.IsNullOrEmpty(searchText)) and the non debounced version if the searchtext is empty or null ?
`searchText is set multiple times before the control becomes visible and interactabl`
Is it called with non empty content ?